app/src/main/java/de/westnordost/streetcomplete/screens/main/map/QuestPinsManager.kt

```diff
@@ -140,11 +141,16 @@ class QuestPinsManager(
     }
 
     private fun updateQuestPins(added: Collection<Quest>, removed: Collection<QuestKey>) {
+        val displayedBBox = lastDisplayedRect?.asBoundingBox(TILES_ZOOM)
+        val addedInView = added.filter { displayedBBox?.contains(it.position) != false }
+        var deletedAny = false
         val pins = synchronized(questsInView) {
-            added.forEach { questsInView[it.key] = createQuestPins(it) }
-            removed.forEach { questsInView.remove(it) }
+            addedInView.forEach { questsInView[it.key] = createQuestPins(it) }
+            removed.forEach { if (questsInView.remove(it) != null) deletedAny = true }
             questsInView.values.flatten()
         }
+        if (!deletedAny && addedInView.isEmpty())
+            return
         pinsMapComponent.set(pins)
```

Comment:
```suggestion
        if (deletedAny || addedInView.isNotEmpty())
            pinsMapComponent.set(pins)
```

Just to make it consistent with the code in StyleableOverlayManager